app/helpers/blacklight/facets_helper_behavior.rb

```diff
@@ -136,20 +136,28 @@ def render_facet_value(facet_field, item, options ={})
   def path_for_facet(facet_field, item)
     facet_config = facet_configuration_for_field(facet_field)
     if facet_config.url_method
-      path = send(facet_config.url_method, facet_field, item)
+      send(facet_config.url_method, facet_field, item)
     else
-      path = search_action_path(add_facet_params_and_redirect(facet_field, item))
+      search_action_path(current_path.add_facet_params_and_redirect(facet_field, item))
     end
   end
 
+  def current_path
```

Comment:
:+1: 

:ok_hand: if it's easy. it doesn't look like the class is much overhead, but it probably gets called a fair bit when rendering facets.

I'll think on it some more, but that shouldn't stop us from shipping.